R/visr.R

```diff
@@ -279,21 +280,21 @@ visr.survfit <- function(
 
 # Obtain Y-asis label ----------------------------------------------------------
 
-  if (is.null(y_ticks) & is.character(fun)){
-    y_ticks <- base::switch(
+  if (is.null(y_ticks) & is.character(fun)) {
+    y_ticks <- switch(
       fun,
       surv = pretty(c(0, 1), 5),
-      log =  pretty(round(c(ymin, ymax), 0), 5),
+      log = pretty(c(ymin, ymax), 5),
       event = pretty(c(0, 1), 5),
-      cloglog = pretty(round(c(ymin, ymax), 0), 5),
+      cloglog = pretty(c(ymin, ymax), 5),
       pct = pretty(c(0, 100), 5),
       logpct = pretty(c(0, 5), 5),
-      cumhaz =  pretty(round(c(ymin, ymax), 0), 5),
+      cumhaz = pretty(c(ymin, ymax), 5),
       stop("Unrecognized fun argument")
     )
   } else if (is.null(y_ticks) & is.function(fun)) {
     
-    y_ticks = pretty(round(c(ymin, ymax), 0), 5)
+    y_ticks = pretty(c(ymin, ymax), 5)
```

Comment:
it needs to be rounded to avoid clutter on the y axis. I agree that pretty is not the best choice. Did you find alternatives?
You could determine the amount of digits and then limit them to amount-1

Reply:
In the end, `pretty` wasn't really the problem since that respects the value range. But, given that we were rounding with `0` precision, this would for example round 1.499999 to 1. 

The alternative approach would be a significantly more elaborate helper function that 1) rounds, 2) pretties and 3) then extend the upper/lower bound my extrapolating with the `pretty` result spacing between numbers to potentially add a 6th/7th before and/or after.